python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -276,10 +277,13 @@ def run(self, splash, version, **kwargs):
         if kwargs.get("server") is None:
             # Initialize all of this after the style-sheet has been applied so any prompt are also
             # styled after the Shotgun Desktop's visual-style.
-            splash.set_message("Initializing browser integration.")
+            if splash:
+                splash.set_message("Initializing browser integration.")
             try:
                 desktop_server_framework = sgtk.platform.get_framework("tk-framework-desktopserver")
-                desktop_server_framework.launch_desktop_server(self._user.host, self._current_login["id"])
+                desktop_server_framework.launch_desktop_server(
+                    self._user.host, self._current_login["id"], parent=splash
```

Comment:
if the splash is `None` or when the splash goes away, will the parenting here cause issues you think? Normally we deallocate in reverse order, e.g. all children are destroyed before the parent - here is suspect the desktop server will live on beyond the life cycle of the splash? In any case, may be worth a comment explaining!